server/server.go

```diff
@@ -519,6 +519,12 @@ func killConn(conn *clientConn, query bool) {
 	}
 
 	conn.mu.RLock()
+	for _, resultSet := range conn.mu.resultSets {
+		// resultSet.Close() is reentrant so it's safe to kill a same connID multiple times
```

Comment:
maybe `tidbResultSet#closed` still have race-condition?

Reply:
Fixed.
Now `TidbResultSet` is thread-safe.